### ISSUE TITLE
 IOU - Re-creating Split Bill with anonymous gmail members causes IOU to crash

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#30140]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.88-1

**Reproducible in staging?:** Yes

**Reproducible in production?:**  No

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

## Action Performed:
1. Open New Expensify app
2. Log in with an expensifail account
3. Create Split Bill with gmail accounts with which there was no conversation history before
4. In the created group DM with these members, request the Split Bill again

## Expected Result:
Re-creating Split Bill with anonymous gmail members should register IOU with no errors for both the re-request and the first

## Actual Result:
Re-creating Split Bill with anonymous gmail members causes IOU to crash

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] Windows: Chrome
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>Windows: Chrome</summary>

 

https://github.com/Expensify/App/assets/78819774/d058135b-bade-4af3-a726-e08bcc53cd5f

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by pradeepmdk** [comment]
### Proposal
### Please re-state the problem that we are trying to solve in this issue.
IOU - Re-creating Split Bill with anonymous gmail members causes IOU to crash
### What is the root cause of that problem?

<img width="1681" alt="Screenshot 2023-10-22 at 7 53 38 AM" src="https://github.com/Expensify/App/assets/16595846/7a213935-1b94-4ff0-ac18-17647ed4408d">
on new user when again split we are calling the at the time data not passing  correctly
<img width="1668" alt="Screenshot 2023-10-22 at 7 56 06 AM" src="https://github.com/Expensify/App/assets/16595846/64465680-daf0-44bc-bf2a-b6da752f74a8">
<img width="781" alt="Screenshot 2023-10-22 at 10 00 22 AM" src="https://github.com/Expensify/App/assets/16595846/71b730e2-a5c4-4bc9-91ba-bcdb72efd9fe">
the root cause when we call `SplitBill` its throwing the error  

### What changes do you think we should make in order to solve the problem?
this is backend issue. 
because first time we re calling `SplitBillAndOpenReport` after that we are personal details with real account id. then we are calling `SplitBill` with those id when create split from report 


onemore issue when we call `SplitBillAndOpenReport` it will return personal details without `login` key.

**Comment by mountiny** [comment]
This is an backend issue so not App deploy blocker technically

**Comment by flodnv** [comment]
@luacmartins @mountiny seems like you got this?

**Comment by mountiny** [comment]
Yes, this PR should fix it https://github.com/Expensify/Web-Expensify/pull/39344

**Comment by pradeepmdk** [comment]
@mountiny i just request again

**Comment by mountiny** [comment]
@youssef-lr was debugging with Youssef and there are some other cases where this can fail due to the optimistic reports/ accountIDs, fix is underway

**Comment by luacmartins** [comment]
PRs were merged and CPed to staging. I think we're good to test again!

**Comment by luacmartins** [comment]
Ah I see! Thanks for clarifying and working on it!

**Comment by youssef-lr** [comment]
Np! Also I edited my comment, without Step 2 this should work fine, so we half fixed this so far by using the split chat report returned from Auth. 

https://github.com/Expensify/App/assets/9680864/bfa0f30c-415d-4dd0-9832-6797d34bd4c0

**Comment by pradeepmdk** [comment]
@youssef-lr looks like one more case https://github.com/Expensify/App/issues/30220 do you think this will fixed here?

**Comment by DylanDylann** [comment]
@ntdiary My suggestion is mentioned in the [proposa](https://github.com/Expensify/App/issues/30220#issuecomment-1782567787)l:

In here:
https://github.com/Expensify/App/blob/eef8347c59b5a3bb7fc508484f4c32ff092a05da/src/libs/OptionsListUtils.js#L1243-L1244
we use ```!reportOption.login``` to check if the report has multiple participants or not, but it also removes the report with NEW account, because its login value is undefined.
So we can check ```null``` rather than just ```!reportOption.login```

**Comment by DylanDylann** [comment]
Yeah. Just want to wrap my suggestion solution for everyone who does not have a context in that issue  @ntdiary

**Comment by mountiny** [comment]
I have not really helped with the fix here, so I am going to unassign myself from this one

**Comment by youssef-lr** [comment]
>we use !reportOption.login to check if the report has multiple participants or not, but it also removes the report with NEW account, because its login value is undefined.

yeah I did come across this, I have a draft PR that fixes it but it also includes completly relying on the login to check for accounts from the search and uses `accountID`, I have one remaining bug to resolve before I can make it ready. I don't think it's as simple as replacing the check with `null`, are there any side effects when you apply that change @DylanDylann?

**Comment by youssef-lr** [comment]
Also in my draft PR, I have made this change to the line you linked:

`if (!isCurrentUserOwnedPolicyExpenseChatThatCouldShow && !includeMultipleParticipantReports && reportOption.participantAccountIDs.length > 1`

**Comment by DylanDylann** [comment]
@youssef-lr my solution is using``` reportOption.login === null``` in stead of ```!reportOption.login```

**Comment by DylanDylann** [comment]
@youssef-lr but i just checked the latest main, it works properly. You can verify it

**Comment by DylanDylann** [comment]
> @youssef-lr but i just checked the latest main, cannot reproduce this bug. You can verify it

Oh. Can reproduce it now

**Comment by DylanDylann** [comment]
@youssef-lr With my proposal, we need to apply the changes in a few sections. But before creating a draft PR, I want to confirm that: Should we fix this issue from BE side? For example, when we create a new chat with a new account, BE will return the report with the ```login``` field

**Comment by stephanieelliott** [comment]
Bump on the above @youssef-lr -- what do you think?

**Comment by youssef-lr** [comment]
>Should we fix this issue from BE side? For example, when we create a new chat with a new account, BE will return the report with the login field

I don't think the backend sends this, have you confirmed it in the Network tab? We set this in `OptionUtils` when creating the optimistic account from the search.

**Comment by DylanDylann** [comment]
@youssef-lr I mean that, currently, when we start a chat with the new account, BE will return the new account`s personalDetail without the login field, which leads to the bug. So I want to confirm that, should we let BE return the login field in personalDetail?

**Comment by marcaaron** [comment]
@ntdiary Can you maybe ask your question another way or add more details to it?

>I also traced back the git history and found that the condition to add the login field was introduced

Which condition exactly?

>I didn't find the exact reason for adding it, so I'm not quite sure if replacing it will cause a regression

Replace it with what?

**Comment by marcaaron** [comment]
Thinking through what the code is doing...

```js
 if (!includeMultipleParticipantReports && !reportOption.login) {
```

```js
!includeMultipleParticipantReports
```
This means don't want to include the multiple participants.

>this report has multiple participants

Apparently we know that because `!reportOption.login`

Sounds weird. Let's look at where things were at when we wrote this...

https://github.com/Expensify/App/blame/c17b9c22c064883eb3ffa1bc622f288dec985433/src/libs/ReportListUtils.js#L106-L108

**Comment by DylanDylann** [comment]
@marcaaron so you use ```reportOption.login``` to check if the report has multiple participants or not, right?

**Comment by marcaaron** [comment]
>if we create a chat with a new account, and then request money from that account, we will fail to retrieve the existing report because !reportOption.login is true

Interesting, why is it `undefined`?

>and then we will send a new optimistic reportID to server, thus causing the request to fail

Sorry, I didn't quite understand this part. Without doing a deeper dive into this issue it sounds like an edge case that we haven't considered because the logic for whether to show group DMs in a selector predates the addition of manual requests.

If we want those group DMs to show up in some other selector then it sounds like we need better logic to tell if a chat has multiple participants. And probably the option could have a boolean like `hasMultipleParticipants` instead of using the `login` (which looking back, I can see is confusing). Maybe it was reliable to use the existence of the `login` at once point. But sounds like we need some other solution now.

Please do let me know if there is a more specific question. Thanks!

**Comment by stephanieelliott** [comment]
Hey @youssef-lr I'm not able to find the draft PR, mind linking it here?

**Comment by youssef-lr** [comment]
I haven't pushed yet, I will try to make a PR this week @stephanieelliott

**Comment by stephanieelliott** [comment]
I'm thinking this belongs on #vip-split, posted in Slack to confirm: https://expensify.slack.com/archives/C05RECHFBEW/p1704325452233169

**Comment by stephanieelliott** [comment]
Removing the `Reviewing` label since that refers to the first PR (https://github.com/Expensify/Web-Expensify/pull/39348). Now working on the followup PR, so we'll be going through the review cycle again!

**Comment by luacmartins** [comment]
I'll unassign myself since @youssef-lr seems to be handling this.

**Comment by stephanieelliott** [comment]
Hey @youssef-lr any update on a PR for this one?

**Comment by youssef-lr** [comment]
I have a PR in the works that's going to partially help solve this, as long as the user doesn't log out. It should be ready for review by EOD Monday.

**Comment by stephanieelliott** [comment]
Hey @youssef-lr is there an update on the PR?

**Comment by youssef-lr** [comment]
It's under review @stephanieelliott. I added a hold on this [issue](https://github.com/Expensify/Expensify/issues/328013), which should also fix this one.

**Comment by stephanieelliott** [comment]
Oop Melvin was kirking out earlier -- this is not overdue!

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/Expensify/issues/328013

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/Expensify/issues/328013

**Comment by youssef-lr** [comment]
Not overdue, remaining PRs in the works.

**Comment by youssef-lr** [comment]
Same as above.

**Comment by stephanieelliott** [comment]
Moving this to `Weekly` to match the K2 of the issue we're held on

**Comment by stephanieelliott** [comment]
Still held

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/Expensify/issues/328013

**Comment by youssef-lr** [comment]
Web-E is close to getting merged, just addressed one final coment.

**Comment by youssef-lr** [comment]
Hmm I thought this would be fixed, but we have a new issue where invited new accounts still show up as "Hidden", with their login not being saved, we're unable to get the existing chat locally so it fails in the backend.


https://github.com/Expensify/App/assets/9680864/45103aa2-19fb-4067-8ad1-3898030f7712

@puneetlath given that I've invited these accounts, shouldn't they be known to me?

**Comment by stephanieelliott** [comment]
Bumping in case the mention got buried under the image -- any thoughts on the above @puneetlath?

**Comment by puneetlath** [comment]
@youssef-lr yes, they should be. Maybe we aren't properly setting the inviterID via the split flow.

**Comment by youssef-lr** [comment]
Thanks @puneetlath, I'll look into that this week.

**Comment by stephanieelliott** [comment]
Hey @youssef-lr we should take this off hold now, yea? 

Also, not sure if it helps but I did a deep dive on `Hidden` account behavior a few weeks ago and shared the notes [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1707901044961849). Obviously doesn't explain the crashing, but it does lay out some scenarios where a user may display as hidden.

**Comment by youssef-lr** [comment]
I think this is definitely due to the 'Hidden' issue, and I agree we can take this off hold. The proper solution would be that any account we start a chat with using their email should never show up as Hidden to us, but the solution to this is not straightforward unfortunately and we haven't decided yet how to fix it, there is a long thread [here](https://expensify.slack.com/archives/C066HJM2CAZ/p1701193182994409) about it.

Another option would be to make the split backend take just accountIDs and not emails, I think this would be more straightforward to implement. I'll dive deeper into this this week and report back here.

**Comment by youssef-lr** [comment]
No update yet.

**Comment by youssef-lr** [comment]
Same. Haven't found time for this yet.

**Comment by stephanieelliott** [comment]
Gonna move this back to Weekly since it's been re-prioiritized to `LOW`

**Comment by stephanieelliott** [comment]
Noting here that I'll be OOO til May 2. I'll leave myself assigned here since there hasn't been much movement, but if you need a BZ before then just reapply the `Bug` label to get someone else and I will pick it back up from them when I return!

**Comment by stephanieelliott** [comment]
We're still holding for #323235

**Comment by youssef-lr** [comment]
Still held

**Comment by stephanieelliott** [comment]
Held now on https://github.com/Expensify/Expensify/issues/357867. I am going to be OOO til July 10 but not expecting much movement on this so will keep it assigned. If it needs attention in the meantime please reapply the BZ label

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/Expensify/issues/357867

**Comment by youssef-lr** [comment]
Still held

**Comment by youssef-lr** [comment]
This is paused

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/Expensify/issues/357867

**Comment by youssef-lr** [comment]
Same

**Comment by youssef-lr** [comment]
Same

**Comment by cead22** [comment]
@davidcardoza should this be on the #billpay project?

**Comment by stephanieelliott** [comment]
Seems like this has been fixed! Could not repro, split expense worked fine with no error or crash: 

<img width="712" alt="image" src="https://github.com/user-attachments/assets/7f1c809b-e8db-4972-b3b6-ce12159fd274">
